### ISSUE TITLE
Duplicate images being stored after processing

### ISSUE THREAD

**Issue by rubyruins** [flow2ml/Flow2ML#48]
# Current Behavior:

Currently, this is the code I have run:

```
# To be given input by the user.
img_dimensions = (150,150,1)
test_val_split = 0.5

# Import flow2ml package
from flow2ml import Flow

# Give the Dataset and Data directories
flow = Flow( 'dataset_dir' , 'data_dir' )

# Define The Filers to be used
filters = ["median", "gaussian", "sobelx", "sobely"]

# Apply The Filters
flow.applyFilters(filters)

(train_x, train_y, val_x, val_y) = flow.getDataset(img_dimensions, test_val_split)
```

I have used 2 classes with 5 images each in for the filters feature. A new folder processedData contains the processed images with the filters applied, however, the processed images are also present in the original folder data_dir containing the class images. **The same image is duplicated and stored in 2 different places.** Screenshot attached below:

# Screenshots (optional):

<p float="left">
  <img src="https://user-images.githubusercontent.com/50259869/114762319-404b0d80-9d7f-11eb-97e1-7f32a2f06346.png" width="300" />
  <img src="https://user-images.githubusercontent.com/50259869/114762660-9d46c380-9d7f-11eb-96e7-5b11d97c8796.png" width="300" /> 
</p>

# Expected Behavior:

The processed images are getting stored in the data_dir for the class, and are duplicates of the images in the processedData folder. It would be more convenient for the user if the processed images are only there in the separate folder processedData, and not in the original folder containing the class images. 

I am also working on the unit tests issue, and it would be easier to test the functionalities if this issue is resolved.

# Solution: 

After the processedData folder is generated, delete the duplicate images and folders from the data_dir folder. Eg: data_dir/apple/GaussianImages, data_dir/apple/MedianImages etc.

**Comment by chebroluharika** [comment]
@yvkrishna Can you please respond here. Is that an expected behavior?

**Comment by yvkrishna** [comment]
yes that happens, While I was working on the code, once I have got an error in between the process then the images were moved to processedData folder and the entire process stopped in middle. so at that time Instead of shifting the images I copied and pasted it in processedData folder.

As said by @rubyruins we can try to delete all Filtered applied images folders like MedianImages, etc after transferring them to processedData folder

or else, instead of saving images in folders, we can try to save images as numpy matrices in python variables.

**Comment by rubyruins** [comment]
If we save the images as numpy matrices, they won't be available outside of the program, right? I think having them in folders could be a better option. As you suggested, after the processedData folder is created and filled, we could delete the corresponding images from the original folder?

**Comment by rubyruins** [comment]
@yvkrishna, what should I do about this, then? Should we remove the duplicate images or close the issue?

**Comment by yvkrishna** [comment]
> If we save the images as numpy matrices, they won't be available outside of the program, right? I think having them in folders could be a better option. As you suggested, after the processedData folder is created and filled, we could delete the corresponding images from the original folder?

we can store it using **self** keyword. With that we can able to store matrices for all the methods in that class.

**Comment by yvkrishna** [comment]
> @yvkrishna, what should I do about this, then? Should we remove the duplicate images or close the issue?

initially try to store all the images in processedData folder and once copying is done, then you can delete the folders inside each labeled folder.

Once the above step is done, try to store the NumPy matrices as a variable as a class variable. and you can return the variables.

**Comment by rubyruins** [comment]
@yvkrishna, oh okay. Yeah, I got the first part about deleting the folders inside data_dir. Where should I store it as the class variable? 
In this line `(train_x, train_y, val_x, val_y) = flow.getDataset( img_dimensions, test_val_split )`, we are storing the image matrices in the 4 variables, right? Should I also store it anywhere else?

**Comment by yvkrishna** [comment]
> @yvkrishna, oh okay. Yeah, I got the first part about deleting the folders inside data_dir. Where should I store it as the class variable?
> In this line `(train_x, train_y, val_x, val_y) = flow.getDataset( img_dimensions, test_val_split )`, we are storing the image matrices in the 4 variables, right? Should I also store it anywhere else?

Actually, I am also not much familiar with storing results in the class variable. But you can create a class variable in __init__ function and try to append image arrays to that variables. @chebroluharika or @jaswaniyogita can you try to help in this case?

coming to the line `(train_x, train_y, val_x, val_y) = flow.getDataset( img_dimensions, test_val_split )`  here under the hood, we read each image from folders and append it to the NumPy matrix. This is okay but as another solution, we can return the class variables too, but I am not proficient in them.

**Comment by rubyruins** [comment]
Oh, okay!

**Comment by subhangi2731** [comment]
please assign me this

**Comment by jaswaniyogita** [comment]
> please assign me this

Please start working on this. I have assigned this to you.

**Comment by chebroluharika** [comment]
@subhangi2731 do we have any update on this?

**Comment by yvkrishna** [comment]
@subhangi2731, Can you please give us update on this ?

**Comment by subhangi2731** [comment]
@yvkrishna I am working on this I will give u an update by today night

**Comment by yvkrishna** [comment]
> @yvkrishna I am working on this I will give u an update by today night

Okay 👍

**Comment by chebroluharika** [comment]
@subhangi2731 can we expect output by tomorrow EOD?

**Comment by chebroluharika** [comment]
As there is no update from @subhangi2731 since 1 week, unassigning it.

**Comment by chebroluharika** [comment]
No @ArkaprabhaChakraborty

**Comment by ArkaprabhaChakraborty** [comment]
I can work on this.

**Comment by yvkrishna** [comment]
@ArkaprabhaChakraborty, can you complete this issue before the program period ends ?

**Comment by ArkaprabhaChakraborty** [comment]
Man the entire code for this is a mixed porridge 🙂

**Comment by ArkaprabhaChakraborty** [comment]
I dont think I can. The code is so fuzzed up its difgicult to spot "what" is happening "where".

**Comment by rubyruins** [comment]
I think some of the subsequent code uses images from data_dir in further processing, and processedData is needed for training. So maybe this issue is not relevant now.

**Comment by chebroluharika** [comment]
Okay closing this.
